pkg/kubelet/api/v1alpha1/runtime/api.proto

```diff
@@ -607,10 +611,32 @@ message ExecRequest {
 }
 
 message ExecResponse {
-    // Streaming stdout
-    optional bytes stdout = 1;
-    // Streaming stderr
-    optional bytes stderr = 2;
+    // The fully qualified URL of the exec streaming server
+    optional string url = 1;
+}
+
+message AttachRequest {
+    // The id of the container
+    optional string container_id = 1;
+    // Streaming stdin
+    optional bytes stdin = 2;
```

Comment:
That's true of all the parameters (except the IDs), but I have a couple reasons for specifying them here:

1. It's explicitly enumerating the options in the API, as opposed to the URL parameters which aren't documented anywhere.
2. It adds more flexibility for alternative implementations, e.g. if a runtime wanted to execute the command before waiting for the connection to the server, or use a token rather than URL parameters.
3. It leaves URL format up to the runtime implementation, since the runtime determines the URL.

Initially, and in most cases, I expect runtime implementations to just encode the parameters in the URL that's returned.

Thoughts?